RecoVertex/BeamSpotProducer/plugins/OnlineBeamSpotESProducer.cc

```diff
@@ -80,33 +84,33 @@ const BeamSpotOnlineObjects* OnlineBeamSpotESProducer::compareBS(const BeamSpotO
 
   // Logic to choose between the two BeamSpots:
   // 1. If both BS are older than limitTime retun fake BS
-  // 2. If only one BS is newer than limitTime return it only if it has
-  //    sigmaZ larger than sigmaZthreshold_ and the fit converged (BeamType 2)
-  // 3. If both are newer than the limit threshold return
-  //    the BS that converged and has larger sigmaZ
+  // 2. If only one BS is newer than limitTime return it only if
+  //     it passes isGoodBS (checks on sigmaZ, sigmaXY and fit convergence)
+  // 3. If both are newer than the limit threshold return the BS that
+  //     passes isGoodBS and has larger sigmaZ
   if (diffBStime1 > limitTime && diffBStime2 > limitTime) {
     edm::LogInfo("OnlineBeamSpotESProducer") << "Defaulting to fake because both payloads are too old.";
     return nullptr;
   } else if (diffBStime2 > limitTime) {
-    if (bs1->sigmaZ() > sigmaZThreshold_ && bs1->beamType() == 2) {
+    if (isGoodBS(bs1)) {
       return bs1;
     } else {
       edm::LogInfo("OnlineBeamSpotESProducer")
           << "Defaulting to fake because the legacy Beam Spot is not suitable and HLT one is too old.";
       return nullptr;
     }
   } else if (diffBStime1 > limitTime) {
-    if (bs2->sigmaZ() > sigmaZThreshold_ && bs2->beamType() == 2) {
+    if (isGoodBS(bs2)) {
       return bs2;
     } else {
       edm::LogInfo("OnlineBeamSpotESProducer")
           << "Defaulting to fake because the HLT Beam Spot is not suitable and the legacy one too old.";
       return nullptr;
     }
   } else {
-    if (bs1->sigmaZ() > bs2->sigmaZ() && bs1->beamType() == 2) {
+    if (bs1->sigmaZ() > bs2->sigmaZ() && isGoodBS(bs1)) {
       return bs1;
-    } else if (bs2->sigmaZ() >= bs1->sigmaZ() && bs2->beamType() == 2) {
+    } else if (bs2->sigmaZ() >= bs1->sigmaZ() && isGoodBS(bs2)) {
       return bs2;
     } else {
       edm::LogInfo("OnlineBeamSpotESProducer")
```

Comment:
the log message that comes after this line could be updated to reflect the change in the logic.

Reply:
Ciao Marco. I changed the log message in "https://github.com/cms-sw/cmssw/pull/40744/commits/1ec86e9f13de1e68d1431e5cb9b7462686106636"